test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -1103,6 +1103,30 @@ def test_torch_save_load(self):
                 else:
                     self.assertEqual(v, state_dict[k])
 
+    @skip_if_lt_x_gpu(2)
+    def test_shared_module(self):
+        class TestDummyModel(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                torch.manual_seed(0)
+                self.net1 = nn.Sequential(nn.Linear(8, 16), nn.ReLU())
+                self.net2 = nn.Sequential(nn.Linear(16, 16), nn.ReLU())
+                self.net3 = self.net2
+                self.random_parameter = nn.Parameter(torch.Tensor(10))
+                self.shared_parameter = self.random_parameter
+
+            def forward(self, x):
+                return self.net3(self.net2(self.net1(x)))
+
+            def get_input(self):
+                return torch.rand(8, 8, device="cuda")
+
+        model = FSDP(TestDummyModel().cuda())
+        with FSDP.state_dict_type(model, StateDictType.FULL_STATE_DICT):
+            state_dict = model.state_dict()
+            for key, param in state_dict.items():
+                print(key)
```

Comment:
nit: Do we want to keep the `print()` in the unit test? Maybe instead we check that the contents of the state dict are as expected, even by hard coding?